### PR TITLE
Add idle conn duration config to telemetry.sh

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Supports new field in https://github.com/sonic-net/sonic-gnmi/commit/258b8870ce01d29bde315fd163c9c8a4c11f921c

##### Work item tracking
- Microsoft ADO **(number only)**: 13468195

#### How I did it

Add new field in telemetry.sh

#### How to verify it

Pipeline

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 1]
```diff
@@ -76,4 +76,12 @@ else
     TELEMETRY_ARGS+=" --threshold 100"
 fi
 
+IDLE_CONN_DURATION=$(echo $GNMI | jq -r '.idle_conn_duration')
+if [[ $IDLE_CONN_DURATION =~ ^[0-9]+$ ]]; then
+    TELEMETRY_ARGS+=" --idle_conn_duration $IDLE_CONN_DURATION"
+else
+    TELEMETRY_ARGS+=" --idle_conn_duration 5"
```
Comment (qiluo-msft) [inline dockers/docker-sonic-telemetry/telemetry.sh:83]: What is the unit of `5`? Please add as code comment.

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 2]
```diff
@@ -76,4 +76,12 @@ else
     TELEMETRY_ARGS+=" --threshold 100"
 fi
 
+IDLE_CONN_DURATION=$(echo $GNMI | jq -r '.idle_conn_duration')
+if [[ $IDLE_CONN_DURATION =~ ^[0-9]+$ ]]; then
+    TELEMETRY_ARGS+=" --idle_conn_duration $IDLE_CONN_DURATION"
+else
```
Comment (qiluo-msft) [inline dockers/docker-sonic-telemetry/telemetry.sh:82]: If invalid value, is it better to report error and exit?

Comment (qiluo-msft) [reply dockers/docker-sonic-telemetry/telemetry.sh:82]: for missing case, it is okay to use default value.

Comment (zbud-msft) [reply dockers/docker-sonic-telemetry/telemetry.sh:90]: Done, did the same for threshold field as well.

File: dockers/docker-sonic-telemetry/telemetry.sh (hunk 1 above)
Comment (zbud-msft) [reply dockers/docker-sonic-telemetry/telemetry.sh]: Done

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 3]
```diff
@@ -69,11 +70,31 @@ else
...
+# Server will handle threshold connections consecutively
 THRESHOLD_CONNECTIONS=$(echo $GNMI | jq -r '.threshold')
 if [[ $THRESHOLD_CONNECTIONS =~ ^[0-9]+$ ]]; then
     TELEMETRY_ARGS+=" --threshold $THRESHOLD_CONNECTIONS"
 else
-    TELEMETRY_ARGS+=" --threshold 100"
+    if [[ $THRESHOLD_CONNECTIONS == "null" ]]; then
+        TELEMETRY_ARGS+=" --threshold 100"
+    else
+        echo "Incorrect threshold value, expecting positive integers"
```
Comment (qiluo-msft) [inline dockers/docker-sonic-telemetry/telemetry.sh:81]: >echo [](http://example.com/codeflow?start=8&length=4)

echo to stderr

File: dockers/docker-sonic-telemetry/telemetry.sh [hunk 4]
```diff
@@ -69,11 +70,31 @@ else
...
+
+# Close idle connections after certain duration (in seconds)
+IDLE_CONN_DURATION=$(echo $GNMI | jq -r '.idle_conn_duration')
+if [[ $IDLE_CONN_DURATION =~ ^[0-9]+$ ]]; then
+    TELEMETRY_ARGS+=" --idle_conn_duration $IDLE_CONN_DURATION"
+else
+    if [[ $IDLE_CONN_DURATION == "null" ]]; then
+        TELEMETRY_ARGS+=" --idle_conn_duration 5"
+    else
+        echo "Incorrect idle_conn_duration value, expecting positive integers"
```
Comment (qiluo-msft) [inline dockers/docker-sonic-telemetry/telemetry.sh:94]: >echo [](http://example.com/codeflow?start=8&length=4)

echo to stderr

**Review by qiluo-msft — DISMISSED** [review]

File: dockers/docker-sonic-telemetry/telemetry.sh (hunk 4 above)
Comment (zbud-msft) [reply dockers/docker-sonic-telemetry/telemetry.sh]: Done

File: dockers/docker-sonic-telemetry/telemetry.sh (hunk 3 above)
Comment (zbud-msft) [reply dockers/docker-sonic-telemetry/telemetry.sh]: Done

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**